### PR TITLE
added error log when containers are present in the entitlements

### PR DESCRIPTION
**Pull request by trapacska**
autoprov (1) (1)
autoprov (1)

### CODE REVIEW

File: autoprovision/bundleidhelper.go [hunk 1]
```diff
@@ -137,7 +140,7 @@ func appIDName(bundleID string) string {
 }
 
 // CreateBundleID ...
-func CreateBundleID(client *appstoreconnect.Client, bundleIDIdentifier string, entitlements Entitlement) (*appstoreconnect.BundleID, error) {
+func CreateBundleID(client *appstoreconnect.Client, bundleIDIdentifier string, _ Entitlement) (*appstoreconnect.BundleID, error) {
```
Comment (godrei) [inline autoprovision/bundleidhelper.go]: why did you make Entitlement a blank parameter? if it is not used, we should just remove this parameter

File: autoprovision/profilehelper.go [hunk 2]
```diff
@@ -52,12 +54,84 @@ func FindProfile(client *appstoreconnect.Client, profileType appstoreconnect.Pro
...
 	bundleIDresp, err := client.Provisioning.BundleID(prof.Relationships.BundleID.Links.Related)
 	if err != nil {
 		return false, err
 	}
-	return CheckBundleIDEntitlements(client, bundleIDresp.Data, entitlements)
+	return CheckBundleIDEntitlements(client, bundleIDresp.Data, projectEntitlements)
+}
+
+func parseRawProfileEntitlements(prof appstoreconnect.Profile) (serialized.Object, error) {
+	b, err := base64.StdEncoding.DecodeString(prof.Attributes.ProfileContent)
```
Comment (godrei) [inline autoprovision/profilehelper.go]: as WriteProfile has this logic too, what if we assign the decode profile content functionality to the appstoreconnect.Profile struct?

appstoreconnect.Profile.Content() ?

File: autoprovision/profilehelper.go [hunk 3]
```diff
@@ -52,12 +54,84 @@ func FindProfile(client *appstoreconnect.Client, profileType appstoreconnect.Pro
...
+	}
+
+	profile, err := profileutil.NewProvisioningProfileInfo(*pkcs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse profile info from pkcs7 content: %s", err)
+	}
+	return serialized.Object(profile.Entitlements), nil
+}
+
+func findMissingContainers(projectEnts, profileEnts serialized.Object) ([]string, error) {
```
Comment (godrei) [inline autoprovision/profilehelper.go]: pls write a unit test

File: main.go [hunk 4]
```diff
@@ -462,6 +477,20 @@ func main() {
 		}
 	}
 
+	if len(containersByBundleID) > 0 {
```
Comment (godrei) [inline main.go]: I would suggest separating this to a new function, to stop increasing this huge main function

File: autoprovision/bundleidhelper.go (hunk 1 above)
Comment (trapacska) [reply autoprovision/bundleidhelper.go]: that was not the point of this task as we have another static check task but ofc you are right

File: main.go (hunk 4 above)
Comment (trapacska) [reply main.go]: I can't. It contains almost only logs and failf

File: autoprovision/profilehelper_test.go [hunk 5]
```diff
@@ -82,3 +85,73 @@ func Test_profileName(t *testing.T) {
...
+
+func Test_findMissingContainers(t *testing.T) {
+	tests := []struct {
+		name        string
+		projectEnts serialized.Object
+		profileEnts serialized.Object
+		want        []string
+		wantErr     bool
+	}{
+		{
```
Comment (godrei) [inline autoprovision/profilehelper_test.go]: - pls add an edge case: `serialized.Object("")`

**Pull request merged**